docs/DesignDocs/SGX_QuoteVerify_Integration.md

```diff
@@ -0,0 +1,488 @@
+the OE SDK host-side plugin library will dynamically detect the presence of QVL
+library and loads it at runtime. In case the QVL library can't be found,
+an error will be returned.
+
+### Add OCALLs to host-side verifier plugin library to support enclave-side SGX ECDSA-p256 quote verification
+As quote verification will be done by DCAP QVL/QvE, for enclave-side plugin,
+OCALLs in host-side verifier plugin library will be to invoke QVL/QvE.
+
+In this proposal, we will add one OCALL `oe_verify_quote_ocall()` in
+`edl/sgx/attestation.edl`.
+The OCALL is used for passing quote buffer and expiration time flag to host side.
+All other relevant logic will be implemented on the host side, in files
+`host/sgx/ocalls.c` and `host/sgx/quote.c`.
+
+Note that currently only ECDSA-p256 quotes are supported, but the OCALL will keep
+`format_id` and `opt_params` for forward compatibility.
+
+```C
+oe_result_t oe_verify_quote_ocall(
+    [in] const oe_uuid_t* format_id,
+    [in, size=opt_params_size] const void* opt_params,
+    size_t opt_params_size,
+    [in, size=quote_size] const void* p_quote,
+    size_t quote_size,
+    oe_datetime_t expiration_check_date,
+    [out] uint32_t *p_collateral_expiration_status,
+    [out] uint32_t *p_quote_verification_result,
+    [in, out, size=qve_report_size] void* p_qve_report_info,
+    size_t qve_report_size,
+    [out, size=supplemental_data_size] void* p_supplemental_data,
+    size_t supplemental_data_size,
+    [out] size_t* p_supplemental_data_size_out);
+```
+
+### Update implementation of existing plugin functions
+
+These plugin functions include `_verify_evidence()`, `_verify_report()`
+and `oe_verify_quote_with_sgx_endorsements()`.
+
+- Update API `oe_verify_quote_with_sgx_endorsements()` implementation as below:
+  - Input parameter doesn't change.
+  - In host-side plugin:
+    - Set input parameter `p_qve_report_info` to NULL, then call host-side
+      function `oe_sgx_verify_quote()`, which will call SGX QVL library to
+      verify quote.
+  - In enclave-side plugin:
+    - Construct structure `sgx_ql_qe_report_info_t` as below, `nonce` and
+      `app_enclave_target_info` are input, qve_report is QvE report which
+      target to application enclave.
+    ```C
+    typedef struct _sgx_ql_qe_report_info_t {
+      sgx_quote_nonce_t nonce;
+      sgx_target_info_t app_enclave_target_info;
+      sgx_report_t qve_report;
+    }sgx_ql_qe_report_info_t;
+    ```
+    - Call OCALL function `oe_verify_quote_ocall()` to host-side plugin library
+      to verify SGX quote.
+    - With all the QvE return values, call new API.
+      `oe_verify_qve_report_and_identity()` (described below) to
+      verify QvE report and identity.
+- In API `_verify_evidence()`, remove function call `oe_get_sgx_endorsements()`,
+  and only call `oe_parse_sgx_endorsements()` if caller provides an endorsement buffer
+- In API `_verify_report()` and relevant internal APIs, such as
```

Comment:
Is this going to break oe_veriy_report()?

Reply:
It's not the same report in oe_verify_report(). This is the Intel(R) QvE output report, which target to user's application enclave, only user's application enclave can verify this report.

Reply:
I think @yentsanglee's point is a little more general than the specific `_verify_report()` function call; the policy enforcement was implicitly integrated into our quote verification logic in the old codepaths, but the delegation of that responsibility to the QVL/QVE means that there is potentially extra work to maintain parity behavior with the old `oe_get_report()`. 

For example, the `__oe_sgx_set_minimum_crl_tcb_issue_date()` is still a valid public API (in bits) even though we hide it from the doxygen now. It would be helpful to have a  section that briefly talks about what behaviors change with `oe_verify_report()` if it just ends up using this converged code path with the `oe_get_evidence()` APIs.